java/src/org/openqa/selenium/remote/service/DriverService.java

```diff
@@ -110,56 +110,22 @@ protected DriverService(
      Duration timeout,
      List<String> args,
      Map<String, String> environment) throws IOException {
-   this.executable = executable.getCanonicalPath();
+   if (this.executable != null) {
```

Comment:
I think this should be `(executable != null)`.  Because  `this.executable` will always be null before it is assigned in this check.

Reply:
Yes, this is true